src/intpack.rs

```diff
@@ -0,0 +1,59 @@
+#[derive(Clone, Copy, Default, Eq, PartialEq, Debug, Hash)]
+pub struct U24(u32);
+
+impl U24 {
+    #[inline(always)]
+    pub const fn get(self) -> u32 {
+        self.0
+    }
+
+    pub const MAX: u32 = 0x00ffffff;
+}
+
+impl TryFrom<u32> for U24 {
+    type Error = &'static str;
+
+    #[inline(always)]
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        if value <= Self::MAX {
+            Ok(Self(value))
+        } else {
+            Err("value must be smaller than 2^24")
+        }
+    }
+}
+
+#[derive(Clone, Copy, Default, Eq, PartialEq, Debug, Hash)]
+pub struct U24nU8(u32);
+
+impl U24nU8 {
+    #[inline(always)]
+    pub const fn a(self) -> U24 {
+        U24(self.0 >> 8)
+    }
+
+    #[inline(always)]
+    pub fn b(self) -> u8 {
+        u8::try_from(self.0 & u32::from(u8::MAX)).unwrap()
+    }
+
+    #[inline(always)]
+    pub fn set_a(&mut self, a: U24) {
+        self.0 = a.get() << 8 | u32::from(self.b());
+    }
+
+    #[inline(always)]
+    pub fn set_b(&mut self, b: u8) {
+        self.0 = self.a().get() << 8 | u32::from(b);
+    }
```

Comment:
The function names `a()` and `b()` are not human friendly. I think it is better to use more semantic names, e.g., `u24()` or `get_u24()` instead of `a()`.

Reply:
@kg86 
Exactly.
But u8 is reserved for the primitive type and `.get_*()` does not follow the [guidelines](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).
Also, when creating a new struct containing three or more integers such as `struct U48nU8nU8(u64)`, the type name cannot be used to identify the index.
I thought `a` and `b` are similar to indices of tuples such as `.0` and `.1`.

Reply:
I see. I understand the philosophy. Yes, it makes sense. The example of `U48nU8nU8` is very useful.

I think using `.0` and `.1` is more consistent slightly since these structures like `U48nU8nU8` are functionally identical to Tuple, and `.0` and `.1` are more familiar to developers. But their difference is minor, and `a` and `b` also make sense well. So my opinion is just information not to encourage modification.